transcriptLoader.py

```diff
@@ -47,9 +49,17 @@ def makeTranscriptData(self, load=True):
         else:
             self.srtFiles = self.validateVideoFiles()
             self.transcript = processSrtFiles(self.srtFiles)
-            self.combinedTranscript = getCombinedTranscripts(
-                self.transcript, self.config.windowSize
-            )
+            self.processedSentences = getSentences(self.transcript)
+            
+            if self.processedSentences is not None:
+                self.combinedTranscript = getCombinedTranscripts(
+                    self.processedSentences, self.config.windowSize
```

Comment:
What is the reason for `processedSentences` be segmented again using  windowSized? I felt `spacy` is already doing it?  
The `else` call seems to be similar with transcript data.

Reply:
It is because the segmented sentences are still just sentences, and for a lot of cases, a speaker will have many sentences in a short duration of time (Something like: Ah. Yes, hello everyone. Is this on?). We want to still clump those smaller sentences into a reasonably sized block of 30s seconds that be parsed by BERTopic to infer what the topic of discussion in it is about. 

`getCombinedTranscripts` does not ever directly segment the middle of a line, even in the raw SRT transcript, it always goes to the end of the line. So in the case of segmenting sentences, it will go to the end of the sentence. Another watyto describe is that the segmentation is done to ensure that a minimum of 30s of dialogue is present in a segment, irrespective of whether it is the raw transcript, or in sentence form. This allows it to work irrespective of what the data it receives looks like.
As an example, let's say a speaker says a sentence that is 35s long. In that case, the segment is the whole sentence, instead of cutting it up into a smaller segment. The next segment after that will continue on after the end of the sentence.

Reply:
Thanks for explaining.